### PR TITLE
Adding SSN to en_US Person Provider

### PR DESCRIPTION
**Pull request by oh4real**
We are using a 3rd party service that validates the format of SSN, so randomNumber(9) failed their validation about 10% of the time - no SSN in the 900s at the moment.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     },
     "require-dev": {
         "phpunit/phpunit": "~4.0",
-        "squizlabs/php_codesniffer": "~1.5"
+        "squizlabs/php_codesniffer": "~2.0"
```
Comment (fzaninotto) [inline composer.json]: Why this change in this PR?

File: src/Faker/Provider/en_US/Person.php [hunk 2]
```diff
@@ -116,4 +116,21 @@ public static function suffix()
...
     }
+
+    /**
+     * @example '444114444' or '444-11-4444'
+     * @param boolean|null $dashed true, false or null
+     */
+    public function ssn($dashed = false)
+    {
+        $ssn = static::randomNumber(9);
+        while (!preg_match('/(?!000)(?!666)(?!9)[0-9]{3}(?!00)[0-9]{2}(?!0000)[0-9]{4}/', $ssn)) {
```
Comment (fzaninotto) [inline src/Faker/Provider/en_US/Person.php]: No, that's not the way to do it, it may take too many tries to get a correct ssn. Use a list of preformatted SSN patterns (like in the `phoneNumber` generator) instead.

Comment (oh4real) [inline src/Faker/Provider/en_US/Person.php]: Howdy,

1/10 (the 9XX-XX-XXXXs) will fail and the extremely rare other cases. We run this routinely and it 90% runs once and 99% runs in two.

I'll look at pre-formatting.

Thanks for feedback!
oh4real

**Comment by fzaninotto** [comment]
No news for 6 months, closing.
